geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/GeoMesaSchemaValidator.scala

```diff
@@ -20,10 +20,31 @@ object GeoMesaSchemaValidator {
   def validate(sft: SimpleFeatureType): Unit = {
     MixedGeometryCheck.validateGeometryType(sft)
     TemporalIndexCheck.validateDtgField(sft)
+    ReservedWordCheck.validateAttributeNames(sft)
   }
 }
 
 /**
+  * Utility object that ensures that none of the (local portion of the) property
+  * names is a reserved word in ECQL.  Using those reserved words in a simple
+  * feature type will cause queries to fail.
+  */
+object ReservedWordCheck extends LazyLogging {
+
+  // ensure that no attribute names are reserved words within GeoTools that will cause query problems
+  def validateAttributeNames(sft: SimpleFeatureType): Unit = {
+    val reservedWords = FeatureUtils.sftReservedWords(sft)
+    if (reservedWords.nonEmpty) {
+      // TODO:  Make this an exception after GeoMesa 1.2.2
```

Comment:
this would only affect new schemas, right? doesn't really matter to me...
so ingesting with the fields itself isn't problematic, it's just if you try to query them, correct?